src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs

```diff
@@ -981,9 +981,10 @@ private Type UseReflectionToGetType(int typeIndex)
             //
             // We don't want to regress the expected case by checking the type info before
             // getting to Type.GetType -- this is costly with v1 resource formats.
-            catch (FileNotFoundException)
+            catch (FileNotFoundException fileNotFoundException)
             {
-                throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
+                // Include the FileNotFoundException as an inner exception to make it more diagnosable
+                throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization, fileNotFoundException);
```

Comment:
Should we also change the message here to indicate something other than not-supported?  For reference here's the source of this exception in .NETFramework.

Looks like this code was previously only throwing `NotSupportedException` on platforms where FEATURE_SERIALIZATION was not set, but now we support serialization.  I wonder if we should bring back the ifdef, but instead as a runtime check of [_permitDeserialization](https://source.dot.net/System.Private.CoreLib/R/40812d5433974bf9.html)

In other words
```c#
if (_permitDeserialization) 
    throw;

throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization, fileNotFoundException);
```

Reply:
Also, it was likely me who missed realizing we needed to undo that ifdef here: https://github.com/dotnet/coreclr/commit/791332b101524034ca9a69ab7669d5a87713447b

I wonder if there are any other interesting `FEATURE_SERIALIZATION` ifdefs that are causing issues.

Reply:
I suppose you mean this? https://github.com/dotnet/coreclr/commit/791332b101524034ca9a69ab7669d5a87713447b#diff-6722a5dffc850c4a6d80bc6880484092e7f159fe79746d04ed74440a16d9f348R764-R767

Reply:
Actually, this won't work for System.Resources.Extensions library. `_permitDeserialization` is only defined in corelib's resource reader, but not on the extensions one. I could ifdef the condition to only be applicable in corelib